### PR TITLE
Feature => Pre-commit hook

### PR DESCRIPTION
**Pull request by sc250024**
Enables this repository to serve as source for a `helm-docs` pre-commit hook compatible with the [Pre-commit](https://pre-commit.com/) project. Alternatively, this could be its own repository; something like `pre-commit-helm-docs`.

Huge shoutout to @antonbabenko and the https://github.com/antonbabenko/pre-commit-terraform project, where I took most of the code from.

## Description

In a nutshell, this enables someone using `pre-commit` to do something like the following in their Helm charts repository:

**.pre-commit-config.yaml**
```yaml
- repo: https://github.com/norwoodj/helm-docs
  rev: master
  hooks:
    - id: helm_docs
```

If there are changes to their Helm chart (which are detected by `helm-docs` itself), this will automatically generate a corresponding `README.md` file. If the user hadn't updated and staged the `README.md` file before, this hook will fail, and ensure that a user *always* updates their `README.md` before they do a commit.

## Motivation and Context
I was always doing this manually before:

```shell
$ cd stable/some-chart
$ helm-docs
$ git add README.md
```

Since I make extensive use of the Pre-Commit project, I wanted a hook to automatically generate the `README.md` for a chart I just changed.

## How Has This Been Tested?
Tested locally. If you want to test a Helm chart repo against this hook, run the following in your Helm charts repo:

```shell
$ pwd
/a/local/copy/of/this/branch/helm-docs

$ cd /some/helm/chart/repo
$ vim stable/mychart/Chart.yaml # Do a quick version bump for testing
$ git add stable/mychart
$ pre-commit try-repo --all-files --verbose '/a/local/copy/of/this/branch/helm-docs' helm_docs
```

You should see that the hook will say "FAIL" because the `README.md` file changed, but it hasn't been staged yet. After staging, the `pre-commit try-repo` command will succeed.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by RothAndrew** [comment]
This would be fantastic! I use pre-commit a lot and this is a perfect fit for it.

File: scripts/helm_docs.sh [hunk 1]
```diff
@@ -0,0 +1,548 @@
+#!/usr/bin/env bash
```
Comment (norwoodj) [inline scripts/helm_docs.sh]: This whole file kinda seems overkill to me. Can't you just check to ensure that helm-docs is installed and then run it? Something like:

```
#!/usr/bin/env bash -e

if ! hash helm-docs; then
    echo "Please install helm-docs to run the pre-commit hook! https://github.com/norwoodj/helm-docs#installation"
    exit 1
fi

helm-docs

File: README.md [hunk 2]
```diff
@@ -174,3 +174,14 @@ configMap:
   not real config param: value
 ```
 
+## Pre-commit hook
```
Comment (norwoodj) [inline README.md]: Thank you for adding docs!

File: .pre-commit-hooks.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
+- id: helm_docs
+  args: []
+  description: Uses 'helm-docs' to create documentation from the Helm chart's 'values.yaml' file, and inserts the result into a corresponding 'README.md' file.
+  entry: scripts/helm_docs.sh
+  exclude: charts\/.*$
+  files: (Chart\.yaml)$
```
Comment (norwoodj) [inline .pre-commit-hooks.yaml]: You will probably want: `(Chart|requirements)\.yaml$`

**Comment by norwoodj** [comment]
Hey @sc250024 thanks a ton for the submission! I had a few comments, wasn't entirely clear why the whole getopt implementation was necessary for the script, but otherwise this is fantastic.

File: .pre-commit-hooks.yaml (hunk 3 above)
Comment (sc250024) [reply .pre-commit-hooks.yaml]: Done

File: scripts/helm_docs.sh (hunk 1 above)
Comment (sc250024) [reply scripts/helm_docs.sh]: @norwoodj This is more of a pre-commit thing, and it's needed to parse the args separate from the file arguments. The BASH `getopt` function is from here => https://github.com/agriffis/pure-getopt

As an example, if I take your BASH script above (with a slight modification of `set -ex` to see debug), and run it with the following YAML file in my branch I've submitted:

```
- id: helm_docs
  args:
    - --log-level
    - debug
  description: Uses 'helm-docs' to create documentation from the Helm chart's 'values.yaml' file, and inserts the result into a corresponding 'README.md' file.
  entry: scripts/helm_docs.sh
  exclude: charts\/.*$
  files: (Chart|requirements)\.yaml$
  language: script
  name: Helm docs
  require_serial: true
```

The result is the following:

```diff
$ pre-commit try-repo --all-files --verbose '/a/local/copy/of/this/branch/helm-docs' helm_docs

[WARNING] Creating temporary repo with uncommitted changes...
===============================================================================
Using config:
===============================================================================
repos:
-   repo: /var/folders/7p/bnb37yz95ld6bnq90blpvqw80000gn/T/tmpvzauqpez/shadow-repo
    rev: 9324abff664cfe1a767a106b140610196d5fa862
    hooks:
    -   id: helm_docs
===============================================================================
[INFO] Initializing environment for /var/folders/7p/bnb37yz95ld6bnq90blpvqw80000gn/T/tmpvzauqpez/shadow-repo.
[helm_docs] Helm docs....................................................Failed
hookid: helm_docs

Files were modified by this hook. Additional output:

+ hash helm-docs
+ helm-docs --log-level debug stable/my-first-chart/requirements.yaml stable/my-other-chart/requirements.yaml stable/my-first-chart/Chart.yaml stable/my-second-chart/Chart.yaml stable/my-other-chart/Chart.yaml stable/my-third-chard/Chart.yaml
time="2019-08-08T13:11:41+02:00" level=debug msg="Ignoring directory .git"
time="2019-08-08T13:11:41+02:00" level=debug msg="Ignoring directory stable/my-other-chart"
time="2019-08-08T13:11:41+02:00" level=info msg="Found Chart directories [stable/common, stable/my-first-chart, stable/my-second-chart, stable/my-third-chard]"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/common"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-third-chard"
time="2019-08-08T13:11:41+02:00" level=debug msg="Did not find template file README.md.gotmpl for chart stable/common, using default template"
time="2019-08-08T13:11:41+02:00" level=debug msg="Did not find template file README.md.gotmpl for chart stable/my-third-chard, using default template"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-first-chart"
time="2019-08-08T13:11:41+02:00" level=info msg="Generating README Documentation for chart stable/my-second-chart"
time="2019-08-08T13:11:41+02:00" level=debug msg="Using template file README.md.gotmpl for chart stable/my-first-chart"
time="2019-08-08T13:11:41+02:00" level=debug msg="Using template file README.md.gotmpl for chart stable/my-second-chart"
```

As you can see, the `pre-commit` binary simply passes the args and the files found via the `files: (Chart|requirements)\.yaml$` regex together to the script in one go. BASH scripts **do**  have argument parsers, and in the end, that's what was being accomplished here.

Anyway, I think the `helm-docs` binary automatically does recursive searching so it should be fine.

File: .pre-commit-hooks.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+- id: helm_docs
+  args: []
+  description: Uses 'helm-docs' to create documentation from the Helm chart's 'values.yaml' file, and inserts the result into a corresponding 'README.md' file.
+  entry: scripts/helm_docs.sh
+  files: (Chart|requirements)\.yaml$
```
Comment (norwoodj) [inline .pre-commit-hooks.yaml]: I'm really sorry, I blanked. Should probably put `values.yaml` in there too, changes to that will affect documentation.

File: scripts/helm_docs.sh [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
```
Comment (norwoodj) [inline scripts/helm_docs.sh]: Also, please rename this file to `git-hook/helm-docs`, and update the `.pre-commit-hooks.yaml` file accordingly. I think it's better to make clearer that this script is only intended for the git hooks.

Comment (sc250024) [reply scripts/helm_docs.sh]: Yup, I will tomorrow, don't have laptop with me tonight.

Comment (sc250024) [reply scripts/helm_docs.sh]: Done.

File: .pre-commit-hooks.yaml (hunk 4 above)
Comment (sc250024) [reply .pre-commit-hooks.yaml]: Updated.

**Comment by sc250024** [comment]
@norwoodj I updated based on the changes you requested.

**Pull request merged**